### ISSUE TITLE
Can't recover more than one never-saved score

### ISSUE THREAD

**Issue by shoogle** [musescore/MuseScore#17238]
### Issue type

Other type of issue

### Bug description

If you create multiple new scores, and MuseScore crashes before you have a chance to save some of them, you will never be able to recover any except the last one you created.

### Steps to reproduce

1. Create a new score (Score 1). Keep it open but don't save it.
2. Create another new score (Score 2).
3. Close Score 2 the normal way, with or without saving.
4. Back in Score 1, go to DevTools > Crash handler > Crash.
5. Launch MuseScore again.

__Expected behaviour:__ MuseScore offers to recover the previous session (i.e. Score 1).
__Actual behaviour:__ MuseScore does not recover Score 1.

**Another variant of this issue:**
In step 3, instead of closing Score 2 the normal way, crash MuseScore and then continue with the other steps.

__Expected behaviour:__ MuseScore offers to recover both sessions (Score 1 and Score 2).
__Actual behaviour:__ MuseScore only offers to recover Score 2.

### Screenshots/Screen recordings

![image](https://user-images.githubusercontent.com/11011881/231844074-24106788-a6de-4e02-9f25-5d339da9e35e.png)

### MuseScore Version

Latest nightly build (95280d8)

### Regression

No.

### Operating system

All

### Additional context

As soon as you create a score, MuseScore immediately creates a backup copy called `new_project.mscz` at the following hidden location:
- **Windows:** `%LOCALAPPDATA%\MuseScore\MuseScore4\`
    - E.g. `C:\Users\USERNAME\AppData\Local\MuseScore\MuseScore4\`
- **macOS:** `~/Library/Application Support/MuseScore/MuseScore4/`
- **Linux:** `~/.local/share/MuseScore/MuseScore4/`

Because this backup file is always called `new_project.mscz`, it overwrites the previous backup file so only the last created score can be recovered. Instead, we should number the backups `new_project1.mscz`, `new_project2.mscz`. or add a creation timestamp to the name like: `new_project_created_2023-04-13-183724.mscz`.

We also **need a design** for the dialog that will appear when more than one backup score is available to recover. The information available for this dialog is:

- The filename of the backup file, which could include the session number or creation date as mentioned above.
- The modification date of the backup file (i.e. the time at which the last autosave occurred).

We could also display information from within the file, such as the score title and thumbnail image, but it might be better to avoid this in case the file is corrupt (e.g. if MuseScore crashed during the autosave operation).

The user should be able to recover all the files and then choose whether or not the keep each of them.

**Comment by shoogle** [comment]
See https://musescore.org/en/node/347446#comment-1185539 and prior discussion.

**Comment by Jap2-0** [comment]
Well... I can confirm this is still an issue.  Note Musescore 3 didn't use "new_project,mscz" but rather a format like "scH15340.mscz" for backups (https://musescore.org/en/node/52116 still says this); I'm curious if that was an issue there.

**Comment by Jap2-0** [comment]
> Note Musescore 3 didn't use "new_project,mscz" but rather a format like "scH15340.mscz" for backups (https://musescore.org/en/node/52116 still says this); I'm curious if that was an issue there.

Yep, [more specifically](https://github.com/musescore/MuseScore/blob/3.6.2/mscore/musescore.cpp#L5218-L5237) MS3 used [Qt's temporary file class](https://doc.qt.io/qt-6/qtemporaryfile.html) to generate a random name.
